lib/matplotlib/colors.py

```diff
@@ -742,6 +742,107 @@ def func_r(x):
 
+    def join(self, other, name=None, frac_self=None, N=None):
+        """
+        Join colormap `self` to `other` and return the new colormap.
+
+        Parameters
+        ----------
+        other : cmap
+            The other colormap to be joined to this one.
+        name : str, optional
+            The name for the reversed colormap. If it's None the
+            name will be ``self.name + '-' + other.name``.
+        frac_self : float in the interval ``(0.0, 1.0)``, optional
+            The fraction of the new colormap that should be occupied
+            by self. By default, this is ``self.N / (self.N +
+            other.N)``.
+        N : int
+            The number of entries in the color map.  The default is ``None``,
+            in which case the number of entries is the sum of the
+            number of entries in the two colormaps to be joined.
+
+        Returns
+        -------
+        LinearSegmentedColormap
+            The joined colormap.
+
+        Examples
+        --------
+        import matplotlib.pyplat as plt
+        cmap1 = plt.get_cmap('jet')
+        cmap2 = plt.get_cmap('plasma_r')
+
+        joined_cmap = cmap1.join(cmap2)
+        """
+        if N is None:
+            N = self.N + other.N
+        if frac_self is None:
+            frac_self = self.N / (other.N + self.N)
+        if name is None:
+            name = '{}+{}'.format(self.name, other.name)
+        assert 0 < frac_self and frac_self < 1, (
+            "The parameter frac_self must be in the interval (0.0, 1.0)."
+        )
+        map0 = self(np.linspace(0, 1, int(N * frac_self)))
+        map1 = other(np.linspace(0, 1, int(N * (1 - frac_self))))
+        # N is set by len of the vstack'd array:
+        return LinearSegmentedColormap.from_list(name, np.vstack((map0, map1)))
+
+    def truncate(self, minval=0.0, maxval=1.0, N=None):
+        """
+        Truncate a colormap.
+
+        Parameters
+        ----------
+        minval : float in the interval ``(0.0, 1.0)``, optional
+            The lower fraction of the colormap you want to truncate
+            (default 0.0).
+
+        maxval : float in the interval ``(0.0, 1.0)``, optional
+            The upper limit of the colormap you want to keep. i.e. truncate
+            the section above this value (default 1.0).
+
+        N : int
+            The number of entries in the map. The default is *None*,
+            in which case the same color-step density is preserved,
+            i.e.:  N = ceil(N * (maxval - minval))
+
+        Returns
+        -------
+        LinearSegmentedColormap
+            The truncated colormap.
+
+        Examples
+        --------
+        import matplotlib.pyplat as plt
+        cmap = plt.get_cmap('jet')
+
+        # This will return the `jet` colormap with the bottom 20%,
+        # and top 30% removed:
+        cmap_trunc = cmap.truncate(0.2, 0.7)
+
+        """
+        assert minval < maxval, "minval must be less than maxval"
```

Comment:
Can you use `if not cond: raise` instead of asserts?   A while ago we went through and removed all of the run-time uses of asserts for input validation as (in principle) the interpreter can decide to ignore the assert statements.